pkg/features/pilot/pilot.go

```diff
@@ -131,6 +131,11 @@ var (
 	// EnableWaitCacheSync provides an option to specify whether it should wait
 	// for cache sync before Pilot bootstrap. Set env PILOT_ENABLE_WAIT_CACHE_SYNC = 0 to disable it.
 	EnableWaitCacheSync = os.Getenv("PILOT_ENABLE_WAIT_CACHE_SYNC") != "0"
+
+	// DisableXDSMarshalingToAny provides an option to disable the "xDS marshaling to Any" feature ("on" by default).
+	DisableXDSMarshalingToAny = func() bool {
+		return len(os.Getenv("PILOT_DISABLE_XDS_MARSHALING_TO_ANY")) != 0
+	}
```

Comment:
since we are checking len, if someone adds any value to this env var, this function would return true which may not be what you wanted.

Reply:
+1 (although, it looks that's the approach used for a lot of disable/enable flags in this file).

Reply:
yes but this is how other features are also implemented. So its a bigger fix..